### PR TITLE
Fixes for release 1.8.0

### PR DESCRIPTION
**Pull request by shaobo-he**
1. Fixed memory allocation procedure `$$alloc` such that it returns
pointers that have range (0, INTMAX) and do not alias.
2. Fixed `bv2int` conversion such that negative bv numbers are correctly
translated.
3. Fixed the heuristic to report the results of memory safety checks.
The new heuristic is more conservative in reporting non-timeout results.
4. Updated Corral version.

### CODE REVIEW

File: lib/smack/SmackRep.cpp [hunk 1]
```diff
@@ -994,8 +1001,21 @@ std::string SmackRep::getPrelude() {
...
+    s << Decl::function(indexedName("$bv2uint",{ptrSizeInBits}), {{"i",bt}}, it, NULL, {Attr::attr("builtin", "bv2int")}) << "\n";
+    const Expr* arg = Expr::id("i");
+    const Expr* uint = Expr::fn(indexedName("$bv2uint", {ptrSizeInBits}), arg);
+    std::string offset;
+    if (ptrSizeInBits == 32)
+      offset = "4294967296";
+    else if (ptrSizeInBits == 64)
+      offset = "18446744073709551616";
+    else
+      llvm_unreachable("Unexpected point bit width.");
```
Comment (zvonimir) [inline lib/smack/SmackRep.cpp]: Should be: Unexpected pointer bit width.

File: lib/smack/SmackRep.cpp [hunk 2]
```diff
@@ -978,7 +978,14 @@ std::string SmackRep::getPrelude() {
...
   s << Decl::axiom(Expr::eq(Expr::id(Naming::GLOBALS_BOTTOM),pointerLit(globalsBottom))) << "\n";
   s << Decl::axiom(Expr::eq(Expr::id(Naming::EXTERNS_BOTTOM),pointerLit(externsBottom))) << "\n";
-  s << Decl::axiom(Expr::eq(Expr::id(Naming::MALLOC_TOP),pointerLit((unsigned long) INT_MAX - 10485760))) << "\n";
+  unsigned long malloc_top;
+  if (ptrSizeInBits == 32)
+    malloc_top = 2147483647UL;
+  else if (ptrSizeInBits == 64)
+    malloc_top = 9223372036854775807UL;
+  else
+    llvm_unreachable("Unexpected point bit width.");
```
Comment (zvonimir) [inline lib/smack/SmackRep.cpp]: Should be: Unexpected pointer bit width.

File: lib/smack/SmackRep.cpp [hunk 3]
```diff
@@ -994,8 +1001,21 @@ std::string SmackRep::getPrelude() {
...
+    const Expr* arg = Expr::id("i");
+    const Expr* uint = Expr::fn(indexedName("$bv2uint", {ptrSizeInBits}), arg);
+    std::string offset;
+    if (ptrSizeInBits == 32)
+      offset = "4294967296";
+    else if (ptrSizeInBits == 64)
+      offset = "18446744073709551616";
+    else
+      llvm_unreachable("Unexpected point bit width.");
+    s << Decl::function(indexedName("$bv2int",{ptrSizeInBits}), {{"i",bt}}, it, Expr::cond(Expr::fn(indexedName("$slt", {bt, "bool"}), {arg, Expr::lit(0UL, ptrSizeInBits)}), Expr::fn(indexedName("$sub", {it}), {uint, Expr::lit(offset, 0U)}), uint), {Attr::attr("inline")});
```
Comment (zvonimir) [inline lib/smack/SmackRep.cpp]: This line is too long. Break it up a bit.

File: share/smack/svcomp/utils.py [hunk 4]
```diff
@@ -423,6 +439,7 @@ def verify_bpl_svcomp(args):
     sys.exit(smack.top.results(args)[result])
 
 def write_error_file(args, status, verifier_output):
+  return
```
Comment (zvonimir) [inline share/smack/svcomp/utils.py]: This seems really hacky. What is happening here?

Comment (shaobo-he) [reply share/smack/svcomp/utils.py]: The C error traces generated by the updated Corral version cannot be correctly converted into SV-COMP witness format. So I just disabled the witness generation.

**Pull request merged**